monitor/engine.go

```diff
@@ -356,3 +356,21 @@ func (e *Engine) Collect(ctx context.Context, levelName string) (*blip.Metrics,
 	atomic.AddUint64(&e.collectFail, 1)
 	return nil, fmt.Errorf("failed to collect %s/%s", e.plan.Name, levelName)
 }
+
+// Stop the engine and cleanup any metrics associated with it.
+func (e *Engine) Stop() {
+	blip.Debug("Stopping engine...")
+	e.planMux.Lock()
+	defer e.planMux.Unlock()
```

Comment:
There's a subtle race condition that we can ignore for now, but let's at least code comment it in this func so future engineers know that we know. If `Engine.Collect` had just been called, then monitor is stopped, it's possible this func locks `planMux` first and blocks the `Engine.Collect` goroutine. Then this func calls the cleanup func that could invalidate something in a collector. After this func returns, `Engline.Collect` can acquire the lock on `planMux` but the cleanup func has already been called, so the collector might crash. But that's ok: the panic will be caught and the monitor is being stopped anywhere. We can just note this potential but unlikely race condition.

Reply:
Would something like having `LevelCollector` wait on the `sem` be sufficient to avoid this? Something like:
```go
case <-stopChan:
   // <other stop code>
   for i := 0; i < maxCollectors; i++ {
	<-sem
   }
   
   c.engine.Stop()
   return nil
```